src/exec.cpp

```diff
@@ -985,22 +985,17 @@ static bool exec_process_in_job(parser_t &parser, process_t *p, job_t *j,
 }
 
 void exec_job(parser_t &parser, job_t *j) {
+    assert(j != nullptr);
```

Comment:
for new assertions we should add an "error message:"

     assert(j != nullptr && "Job should not be null");